src/Symfony/Component/HttpFoundation/Response.php

```diff
@@ -1173,7 +1119,14 @@ public function isNotFound()
      */
     public function isRedirect($location = null)
     {
-        return in_array($this->statusCode, array(201, 301, 302, 303, 307, 308)) && (null === $location ?: $location == $this->headers->get('Location'));
+        return in_array($this->statusCode, array(
+            self::HTTP_CREATED,
+            self::HTTP_MOVED_PERMANENTLY,
+            self::HTTP_FOUND,
+            self::HTTP_SEE_OTHER,
+            self::HTTP_TEMPORARY_REDIRECT,
+            self::HTTP_PERMANENTLY_REDIRECT
```

Comment:
@shrikeh even if we used constants as keys in the array, this would not solve the bug in your imaginary use case. Browsers will react to the status code, not the status text. So if the status code value is wrong, there is nothing which can make things recover.